### PR TITLE
Move nested parameter doc to fragment

### LINKED ISSUE #447: Make common parameter docs fragments?
I was talking to a user on irc today (hi strangeglyph!) and they reported that the use of operating system as a param to other modules is confusing (is it name/description/whatever?). While this might be an issue on its own (strangeglyph promised to submit wording fixes), I thought we might want to move certain params (org, loc, os) to own doc fragments and thus make it easier to update their wording across-the-board?

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #447

### CODE REVIEW

**Comment by evgeni** [comment]
I think we can move more :D

```
plugins/modules/foreman_ptable.py:
    os_family:
      description: The OS family the template shall be assigned with.
plugins/modules/foreman_installation_medium.py:
    os_family:
      description: The OS family the template shall be assigned with. If no os_family is set but a operatingsystem, the value will be derived from it.
plugins/modules/foreman_operatingsystem.py:
    family:
      description:
        - distribution family of the Operating System
```

(the one in operatingsystem is probably not foldable into the other two because of the other name)

File: plugins/doc_fragments/foreman_parameters.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# (c) 2019, Matthias Dellweg <dellweg@atix.de>
```
Comment (evgeni) [inline plugins/doc_fragments/foreman_parameters.py]: you can have multiple fragments in one file, which I think we can use here?

https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fragments

**Comment by mdellweg** [comment]
> I think we can move more :D

I'm not sure, it is worth the trouble there (os_family). With parameters, that corresponds nicely to the shared code that handles the nested entities.

File: plugins/doc_fragments/foreman_parameters.py (hunk 1 above)
Comment (mdellweg) [reply plugins/doc_fragments/foreman_parameters.py]: TIL (and it's still quite early...)

**Comment by evgeni** [comment]
> > I think we can move more :D
> 
> I'm not sure, it is worth the trouble there (os_family). With parameters, that corresponds nicely to the shared code that handles the nested entities.

are you saying we don't have the list of supported OSes shared? ;-)

https://github.com/theforeman/foreman-ansible-modules/blob/87781573d571f7e89eab20feeaddb55a92991878/plugins/module_utils/foreman_helper.py#L747-L763

File: plugins/doc_fragments/foreman.py [hunk 2]
```diff
@@ -45,3 +45,38 @@ class ModuleDocFragment(object):
     default: true
     type: bool
 '''
+
+    NESTED_PARAMETERS = '''
+options:
+  parameters:
+    description:
```
Comment (evgeni) [inline plugins/doc_fragments/foreman.py]: I think if you have `description` set here *and* in the individual modules, you end up with both descriptions joined. Which is not what we want.

Comment (mdellweg) [reply plugins/doc_fragments/foreman.py]: I tested, and at least ansible-doc only takes the one from the module. So i let it be a fallback description.

**Comment by mdellweg** [comment]
> > > I think we can move more :D
> > 
> > 
> > I'm not sure, it is worth the trouble there (os_family). With parameters, that corresponds nicely to the shared code that handles the nested entities.
> 
> are you saying we don't have the list of supported OSes shared? ;-)

OIC. Well...

File: plugins/doc_fragments/foreman.py (hunk 2 above)
Comment (evgeni) [reply plugins/doc_fragments/foreman.py]: ok, then my memory is playing games with me. thanks for testing.

**Pull request merged**